tools/CoqMakefile.in

```diff
@@ -33,7 +33,13 @@ COQLIBS           := $(COQMF_COQLIBS)
 COQLIBS_NOML      := $(COQMF_COQLIBS_NOML)
 CMDLINE_COQLIBS   := $(COQMF_CMDLINE_COQLIBS)
 LOCAL             := $(COQMF_LOCAL)
-COQLIB            := $(COQMF_COQLIB)
+# We add a workaround for #8731; if COQLIB is set externally, don't
+# override it.
```

Comment:
This is not what the line below does. It takes coqlib from a coqtop available in the path whenever COQMF_COQLIB is empty (I've no idea when this is the case).

Reply:
The case happens as described in the bug report.

Indeed I am far from sure this is the right thing to do here.

Reply:
The point is that the comment doesn't match the code.

Reply:
That is, it sounds like we preserve environment COQLIB unconditionally but we only do so when COQMF_COQLIB is unset.